src/notation/view/percussionpanel/percussionpanelpadlistmodel.cpp

```diff
@@ -79,6 +84,18 @@ void PercussionPanelPadListModel::deleteRow(int row)
     emit numPadsChanged();
 }
 
+void PercussionPanelPadListModel::removeEmptyRows()
+{
+    const int lastRowIndex = numPads() / NUM_COLUMNS - 1;
+    for (int i = lastRowIndex; i >= 0; --i) {
+        if (rowIsEmpty(i)) {
+            m_padModels.remove(i * NUM_COLUMNS, NUM_COLUMNS);
```

Comment:
Small improvement/optimization:

1. add a flag rowsRemoved
2. set it here to true
3. emit the signals only if it's true